Sources/BSON/BSONDocument.swift

```diff
@@ -339,7 +303,53 @@ extension BSONDocument: BSONValue {
     }
 
     internal func write(to buffer: inout ByteBuffer) {
-        var doc = ByteBuffer(self._buffer.readableBytesView)
+        var doc = ByteBuffer(self.storage.buffer.readableBytesView)
         buffer.writeBuffer(&doc)
     }
 }
+
+internal struct BSONDocumentStorage {
```

Comment:
this isn't used outside of this file, so it can just be `private`.

Also, it could use a docstring describing its purpose. something like "A wrapper around a `ByteBuffer` providing various `BSONDocument`-specific utilities."